packages/netlify-cms-core/src/actions/config.js

```diff
@@ -145,6 +145,31 @@ export function mergeConfig(config) {
   return { type: CONFIG_MERGE, payload: config };
 }
 
+export async function detectProxyServer(localBackend) {
+  if (location.hostname === 'localhost' || location.hostname === '127.0.0.1') {
+    let proxyUrl;
+    if (localBackend === true) {
```

Comment:
This was going to be my exact next request after reading the test setup with the proxy url specified, nice 👍 👍